### PR TITLE
feat(Api): Implemented SCUI Api

### PR DESCRIPTION
**Pull request by BugisoftRSG**
+ RID Join now allows joining via name

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -38,7 +39,7 @@ BOOL APIENTRY DllMain(HMODULE hmod, DWORD reason, PVOID)
 		g_hmodule = hmod;
 		g_main_thread = CreateThread(nullptr, 0, [](PVOID) -> DWORD
 		{
-			while (!FindWindow("grcWindow", "Grand Theft Auto V"))
+			while (!FindWindow("grcWindow", NULL))
```
Comment (Yimura) [inline src/main.cpp]: Do not use NULL it's a Microsoft extension, prefer `nullptr` or alternatively if `nullptr` isn't accepted use `0`.

File: src/views/network/view_session.cpp [hunk 2]
```diff
@@ -15,11 +15,18 @@ namespace big
...
 		{
 			session::join_by_rockstar_id(rid);
 		});
 
+		static char username[20];
+		ImGui::InputText("Input Username", username, sizeof(username));
+		components::button("Join by Username", []
+		{
+			session::join_by_username(username);
+		});
```
Comment (Yimura) [inline src/views/network/view_session.cpp]: Probably best to use `g_thread_pool::push()` as the API lookup will definitely stall the game.

**Review by Yimura — CHANGES_REQUESTED** [review]

**Review by Yimura — DISMISSED** [review]

File: src/views/network/view_session.cpp [hunk 3]
```diff
@@ -16,11 +16,18 @@ namespace big
...
 		{
 			session::join_by_rockstar_id(rid);
 		});
 
+		static char username[20];
+		ImGui::InputText("Input Username", username, sizeof(username));
+		if (components::button("Join by Username"))
+		{
+			session::join_by_username(username);
+    };
```
Comment (tupoy-ya) [inline src/views/network/view_session.cpp]: Weird indent.

Comment (Yimura) [reply src/views/network/view_session.cpp]: That's my bad, I was fixing the PR in browser.

**Review by Yimura — DISMISSED** [review]

**Review by Yimura — APPROVED** [review]

**Pull request merged**